crates/rpc/src/handlers/validator.rs

```diff
@@ -424,3 +430,61 @@ fn check_validator_participation(
         Ok(validator.is_active_validator(epoch))
     }
 }
+#[get("/validator/attestation_data")]
+pub async fn get_attestation_data(
+    db: Data<ReamDB>,
+    opertation_pool: Data<Arc<OperationPool>>,
+    query: Query<AttestationQuery>,
+) -> Result<impl Responder, ApiError> {
+    let store = Store {
+        db: db.get_ref().clone(),
+        operation_pool: opertation_pool.get_ref().clone(),
+    };
+
+    if store.is_syncing().map_err(|err| {
+        ApiError::InternalError(format!("Failed to check syncing status, err: {err:?}"))
+    })? {
+        return Err(ApiError::UnderSyncing);
+    }
+
+    let slot = query.slot;
+
+    let current_slot = store
+        .get_current_slot()
+        .map_err(|err| ApiError::InternalError(format!("Failed to slot_index, error: {err:?}")))?;
+
+    if slot > current_slot + 1u64 {
+        return Err(ApiError::InvalidParameter(format!(
+            "Slot {slot:?} is too far ahead of the current slot {current_slot:?}"
+        )));
+    }
+
+    let beacon_block_root = db
+        .slot_index_provider()
+        .get_highest_root()
+        .map_err(|err| ApiError::InternalError(format!("Failed to slot_index, error: {err:?}")))?
+        .ok_or(ApiError::NotFound(
+            "Failed to find highest block root".to_string(),
+        ))?;
+
+    let source = db.justified_checkpoint_provider().get().map_err(|err| {
+        ApiError::InternalError(format!("Failed to get source checkpoint, error: {err:?}"))
+    })?;
+
+    let target = db
+        .unrealized_justified_checkpoint_provider()
+        .get()
+        .map_err(|err| {
+            ApiError::InternalError(format!("Failed to target checkpoint, error: {err:?}"))
+        })?;
+
+    let data = AttestationData {
+        slot,
+        index: ELECTRA_COMMITTEE_INDEX,
+        beacon_block_root,
+        source,
+        target,
+    };
+
+    Ok(HttpResponse::Ok().json(DataResponse::new(data)))
```

Comment:
```suggestion
    Ok(HttpResponse::Ok().json(DataResponse::new(AttestationData {
        slot,
        index: ELECTRA_COMMITTEE_INDEX,
        beacon_block_root,
        source,
        target,
    })))
```

this should just be inlined, as the `data` variable isn't used, so there isn't much point in declaring a left value for it